cli/azd/pkg/azsdk/zip_deploy_client.go

```diff
@@ -65,6 +65,12 @@ func NewZipDeployClient(
 	// We do not have a Resource provider to register
 	options.DisableRPRegistration = true
 
+	// Increase default retry attempts from 3 to 4 as zipdeploy often fails with 3 retries.
+	// With the default azcore.policy options, this introduces up to 20 seconds of exponential back-off.
+	options.Retry = policy.RetryOptions{
+		MaxRetries: 4,
+	}
```

Comment:
We could also try adding a longer wait on the first error?  But 4 times is also good

Reply:
It's always a tradeoff. If we delay the early initial retry, we end up being slower on requests that might succeed due to a temporary flake.